runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/jobsubmission/InMemoryJobService.java

```diff
@@ -330,6 +335,29 @@ public void describePipelineOptions(
     }
   }
 
+  @Override
+  public void getJobMetrics(
+      GetJobMetricsRequest request, StreamObserver<GetJobMetricsResponse> responseObserver) {
+    String invocationId = request.getJobId();
+    LOG.info("Running getJobMetrics for {}", invocationId);
+    try {
+      JobInvocation invocation = getInvocation(invocationId);
+      LOG.info("Found job invocation for metrics: {}", invocation);
+      MetricResults metrics = invocation.getMetrics();
```

Comment:
This code is converting MetricResults -> MonitoringInfos -> MetricResults

We really just need the MonitoringInfos to put on the GetJobMetricResponse.
Let's make the interface on JobInvocation return MonitoringInfos. If existing FLink or Samza code already packages them into MetricResults, then they can convert them to MonitoringInfos there.

Ideally that MetricResults format should only be used in that user level API, and not for internals.

Reply:
PipelineResult exposes the SDK-style MetricResults; JobInvocation gets them from there.

Switching to (or adding) MetricResults proto there seemed a bit more involved, but I'll think about it, it might work.

I'd suggest we leave it for a follow-on PR though. This is a case where I tried to play nicely with what the SDK was already doing. We can push for more MI-only APIs everywhere next…

Reply:
@ryan-williams @ajamato I'd like to get more context here. LMK if I'm missing something.  

Here the `JobInvocation` uses `PipelineResult` that contains a `MetricsResult` rather than `MonitoringInfos`. What would be the best way of getting `MonitoringInfos` in JobInvocation?

Is implementing a `MonitoringInfo portableMetrics()` method in `JobServicePipelineResult` and setting `FutureCallback` with `JobServicePipelineResult instead of `PipelineResult` a good idea? We could use it to call: 

`MonitoringInfo metrics = pipelineResult.portableMetrics();`

instead of : 

`MetricResults metrics = pipelineResult.metrics();`

I don't like that we'd have 2 separate methods for `metrics()` both returning different types but currently I don't see other way to 
> make the interface on JobInvocation return MonitoringInfos

WDYT?

Reply:
(@up nevermind - I now see that it is a more complex problem than I initially thought. )

Reply:
Where did you run into trouble with it @lgajowy?

I thought some version of what you described could make sense.

Reply:
Sorry I kept you waiting - I was away. I answered on the devlist, here: https://lists.apache.org/thread.html/70803bd35b0b9b6e1202a3baa9872d140492c354e248de07d5d177f6@%3Cdev.beam.apache.org%3E